mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,660 @@
+            the video pool is empty, it will fetch from the dataset pool
+            (collect proposal of the entire dataset).
+
+            Args:
+                proposal_type (int): Type id of proposal.
+                    Positive/Foreground: 0
+                    Negative:
+                        Incomplete: 1
+                        Background: 2
+                video_id (str): Name of the video.
+                video_pool (list): Proposals in this video.
+                requested_num (int): Number of proposals to be sampled.
+                dataset_pool (list): Proposals of the entire dataset.
+            """
+
+            if len(video_pool) == 0:
+                return [(dataset_pool[x], proposal_type)
+                        for x in np.random.choice(
+                            len(dataset_pool), requested_num, replace=False)]
+            else:
+                replicate = len(video_pool) < requested_num
+                idx = np.random.choice(
+                    len(video_pool), requested_num, replace=replicate)
+                return [((video_id, video_pool[x]), proposal_type)
+                        for x in idx]
+
+        out_props = []
+        out_props.extend(
+            sample_video_proposals(0, record['video_id'], fg,
+                                   self.fg_per_video, self.fg_pool))
+        out_props.extend(
+            sample_video_proposals(1, record['video_id'], incomp,
+                                   self.incomplete_per_video,
+                                   self.incomp_pool))
+        out_props.extend(
+            sample_video_proposals(2, record['video_id'], bg,
+                                   self.bg_per_video, self.bg_pool))
+
+        return out_props
+
+    def _random_sampling(self):
+        """Randomly sample proposals from the entire dataset."""
+        out_props = []
+
+        out_props.extend([(x, 0) for x in np.random.choice(
+            self.fg_pool, self.fg_per_video, replace=False)])
+        out_props.extend([(x, 1) for x in np.random.choice(
+            self.incomp_pool, self.incomplete_per_video, replace=False)])
+        out_props.extend([(x, 2) for x in np.random.choice(
+            self.bg_pool, self.bg_per_video, replace=False)])
+
+        return out_props
+
+    def _get_stage(self, proposal, num_frames):
+        """Fetch the scale factor of starting and ending stage and get the
+        stage split.
+
+        Args:
+            proposal (object): Proposal instance.
+            num_frames (int): Total frames of the video.
+        Returns:
+            starting_scale_factor (float): The ratio of the effective
+                sampling length to augment length in starting stage.
+            ending_scale_factor (float):The ratio of the effective
+                sampling length to augment length in ending stage.
+            stage_split (list): Ending segment id of starting,
+                course and ending stage.
```

Comment:
For docstring of returned vars, you can refer to [this](https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/recognizers/base.py#L101-L104)

Reply:
pls refet to [this](https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/recognizers/base.py#L101-L104) to change the doctring for returned params.